internal/contracts/contracts.go

```diff
@@ -274,4 +275,53 @@ func (mc *JSONContract) Unmarshal() (Contract, error) {
 		mc.StateNonce,
 	}
 	return c, nil
+	fmt.print(c)
+}
+
+func (c *Contract) ContainsPublicKey(pk publickey.AurumPublicKey) (bool, error) {
+	if c == nil {
+		return false, errors.New("not nill")
+	}
+	
+	// encode contract's sender pub key to bytes
+	bKey, err := publickey.Encode(c.SenderPubKey)
+	if err != nil{
+		return bytes.Equal(bKey, pk.Bytes), errors.New("failed to encode sender public key")
+	}
+	// return the comparison of the above encoding with Bytes field of pk
+    return  bytes.Equal(bKey, pk.Bytes), nil
+	
+}
+
+func(c *Contract) GenerateRandomContract() (Contract){
```

Comment:
- [ ] I don't believe we need a receiver, remove it
- [ ] Remove parentheses from return type